To test whether the male-female buyer ratios are similar across regions,we will use the chi-square test for independence.

- Null Hypothesis (H0): Male-female buyer ratios are similar across regions.

- Alternative Hypothesis (H1): Male-female buyer ratios are not similar across regions.

In [10]:
#Contigency Table

#Observed Values	East	West	North	South
#Males	        50	    142	    131	    70
#Females	       435	    1523	1356	750

In [11]:
import pandas as pd
import scipy.stats as stats

In [12]:
# Load the dataset
data=pd.read_csv('BuyerRatio.csv')
data.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [13]:
# Extract the observed values
observed_values = data.iloc[:, 1:].values

print(observed_values)

[[  50  142  131   70]
 [ 435 1523 1356  750]]


In [14]:
# Perform chi-square test
chi2, p_value, _, _ = stats.chi2_contingency(observed_values)

print(f'Chi-square statistic: {chi2}')
print(f'P-value: {p_value}')

# Check significance at the 0.05
alpha = 0.05
if p_value < alpha:
    print('Reject the null hypothesis. Male-female buyer ratios are not similar across regions.')
else:
    print('Fail to reject the null hypothesis. Male-female buyer ratios are similar across regions.')


Chi-square statistic: 1.595945538661058
P-value: 0.6603094907091882
Fail to reject the null hypothesis. Male-female buyer ratios are similar across regions.


The P-value of the Chi-square test is 0.66

Since the p-value from the Chi-square test is more than the significance level, We can say that we fail to reject the null hypothesis.

Hence, Male-female buyer ratios are similar across regions.